rest/server_context.go

```diff
@@ -685,10 +682,19 @@ func dbcOptionsFromConfig(sc *ServerContext, config *DbConfig, dbName string) (d
 		compactIntervalSecs = uint32(*config.CompactIntervalDays * 60 * 60 * 24)
 	}
 
-	resyncQueryLimit := db.DefaultResyncQueryLimit
-	if config.ResyncQueryLimit != nil {
-		resyncQueryLimit = *config.ResyncQueryLimit
+	queryPaginationLimit := db.DefaultQueryPaginationLimit
+	if config.QueryPaginationLimit != nil {
+		queryPaginationLimit = *config.QueryPaginationLimit
+	} else {
+		if config.CacheConfig != nil && config.CacheConfig.ChannelCacheConfig != nil && config.CacheConfig.ChannelCacheConfig.DeprecatedQueryLimit != nil {
+			queryPaginationLimit = *config.CacheConfig.ChannelCacheConfig.DeprecatedQueryLimit
```

Comment:
- Add logging if the DeprecatedQueryLimit is specified.  
- In the case QueryPaginationLimit and DeprecatedQueryLimit are both specified, should also log to indicate that Deprecated is being ignored
- Ensure we're matching previous behaviour if DeprecatedQueryLimit is set to zero (use default)